python/ray/_private/services.py

```diff
@@ -191,7 +191,15 @@ def get_address_info_from_redis_helper(redis_address,
             break
     if relevant_client is None:
         raise RuntimeError(
-            "Redis has started but no raylets have registered yet.")
+            f"This node has an IP address of {node_ip_address}, and Ray "
+            "expects this IP address to be either the Redis address or one of"
+            " the Raylet addresses. Connected to Redis at {redis_address} and"
+            " found raylets at "
+            "{', '.join(c['NodeManagerAddress'] for c in client_table)} but "
```

Comment:
I think you need to have f in the beginning of the string here. 
```suggestion
            f"{', '.join(c['NodeManagerAddress'] for c in client_table)} but "
```

Reply:
Grraahh, yes, you're right. It worked when it was a .format() call, but f-strings have different rules. I'm going to click to accept all your suggestions here, I think it's set up to auto-cancel builds so this shouldn't cause three builds.